packages/apps/google-analytics/src/google-analytics-events.ts

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable no-undef */
+/* eslint-disable import/prefer-default-export */
+import '@cloudcommerce/firebase/lib/init';
+import type { Orders } from '@cloudcommerce/types';
+import type { EventItem, EventParams } from '../types';
+import { getFirestore } from 'firebase-admin/firestore';
+import {
+  createAppEventsFunction,
+  ApiEventHandler,
+} from '@cloudcommerce/firebase/lib/helpers/pubsub';
+import logger from 'firebase-functions/logger';
+import axios from 'axios';
+
+const Axios = axios.create({
+  baseURL: 'https://www.google-analytics.com',
+  headers: {
+    'Content-Type': 'application/json',
+  },
+});
+
+const getSentEventFirestore = (
+  collectionEventsSent: FirebaseFirestore.CollectionReference<FirebaseFirestore.DocumentData>,
+  orderId: string,
+): Promise<FirebaseFirestore.DocumentData | undefined | false> => new Promise((resolve) => {
+  const subscription = collectionEventsSent.doc(orderId);
+  subscription.get()
+    .then((documentSnapshot) => {
+      if (documentSnapshot.exists) {
+        resolve(documentSnapshot.data());
+      } else {
+        resolve(false);
+      }
+    })
+    .catch(() => {
+      resolve(false);
+    });
+});
+
+const handleApiEvent: ApiEventHandler = async ({
+  evName,
+  apiEvent,
+  apiDoc,
+  app,
+}) => {
+  const resourceId = apiEvent.resource_id;
+  logger.info('>> ', resourceId, ' - Action: ', apiEvent.action);
+  const key = `${evName}_${resourceId}`;
+  const appData = { ...app.data, ...app.hidden_data };
+  if (
+    Array.isArray(appData.ignore_events)
+    && appData.ignore_events.includes(evName)
+  ) {
+    logger.info('>> ', key, ' - Ignored event');
+    return null;
+  }
+  logger.info(`> Webhook ${resourceId} [${evName}]`);
+
+  const measurementId = appData.measurement_id;
+  const apiSecret = appData.api_secret;
+
+  const order = apiDoc as Orders;
+  const orderId = order._id;
+
+  const buyer = order.buyers && order.buyers[0];
+  const clientIp = order.client_ip;
+
+  const enabledCustonEvent = order.financial_status?.current && appData.custom_events;
+  const enabledRefundEvent = order.status === 'cancelled' && appData.refund_event;
```

Comment:
Detalhe importante também:

```suggestion
  const enabledRefundEvent = order.status === 'cancelled' && appData.refund_event !== false;
```

https://github.com/ecomplus/cloud-commerce/pull/102#discussion_r1065977343
O refund deve ser enviado por padrão, então ele só não será enviado se a config explicitamente estiver desabilitando, `if (appData.refund_event)` é diferente de `if (appData.refund_event !== false)` ...